apps/pie-storybook/stories/testing/pie-toast.test.stories.ts

```diff
@@ -0,0 +1,209 @@
+import { html } from 'lit';
+import { ifDefined } from 'lit/directives/if-defined.js';
+import { useArgs } from '@storybook/preview-api';
+import { action } from '@storybook/addon-actions';
+import { type Meta } from '@storybook/web-components';
+import { type ToastProps, defaultProps, variants } from '@justeattakeaway/pie-toast';
+
+import {
+    type TemplateFunction,
+    createStory,
+    createVariantStory,
+    type PropDisplayOptions,
+} from '../../utilities';
+
+type ToastStoryMeta = Meta<ToastProps>;
+
+const defaultArgs: ToastProps = {
+    ...defaultProps,
+    message: 'Menu Published',
+    leadingAction: {
+        text: '',
+        ariaLabel: '',
+    },
+    duration: null,
+};
+
+const toastStoryMeta: ToastStoryMeta = {
+    title: 'Toast',
+    component: 'pie-toast',
+    argTypes: {
+        isOpen: {
+            description: 'When true, the toast is set to be open and visible.',
+            control: 'boolean',
+            defaultValue: {
+                summary: defaultProps.isOpen,
+            },
+        },
+        variant: {
+            description: 'Set the variant of the toast.',
+            control: 'select',
+            options: variants,
+            defaultValue: {
+                summary: defaultProps.variant,
+            },
+        },
+        isStrong: {
+            description: 'When true, the toast is displayed with a strong visual style.',
+            control: 'boolean',
+            defaultValue: {
+                summary: defaultProps.isStrong,
+            },
+        },
+        isDismissible: {
+            description: 'Allows dismissing the toast by clicking on the close button.',
+            control: 'boolean',
+            defaultValue: {
+                summary: defaultProps.isDismissible,
+            },
+        },
+        message: {
+            description: 'The message content of the toast.',
+            control: 'text',
+        },
+        isMultiline: {
+            description: 'Allows the message content to be displayed as multiline, limited to three rows.',
+            control: 'boolean',
+            defaultValue: {
+                summary: defaultProps.isOpen,
+            },
+        },
+        leadingAction: {
+            description: 'The leading action configuration for the toast.',
+            control: 'object',
+        },
+        duration: {
+            description: 'Sets the duration of the toast in milliseconds before it auto-dismisses.',
+            control: 'number',
+        },
+    },
+    args: defaultArgs,
+};
+
+export default toastStoryMeta;
+
+const pieToastLeadingActionClick = action('pie-toast-leading-action-click');
+const pieToastClose = action('pie-toast-close');
+const pieToastOpen = action('pie-toast-open');
```

Comment:
Are these being used / tested?

Reply:
Looks like they are used in the controls but maybe we don't need  them in the test story? Or I can write some tests to test that?